dcpy/connectors/edm/recipes.py

```diff
@@ -173,3 +290,80 @@ def purge_recipe_cache():
     """Delete locally stored recipe files."""
     logger.info(f"Purging local recipes from {LIBRARY_DEFAULT_PATH}")
     shutil.rmtree(LIBRARY_DEFAULT_PATH)
+
+
+def log_metadata(config: Config):
+    logger.info(f"Logging library run metadata for dataset {config.dataset.name}")
+    assert (
+        config.execution_details
+    ), f"Provided config for dataset {config.dataset.name} does not have run details specified. Cannot log."
+    pg_client = postgres.PostgresClient(database=LOGGING_DB, schema=LOGGING_SCHEMA)
+    query = f"""
+        INSERT INTO {LOGGING_SCHEMA}.{LOGGING_TABLE_NAME} (name, version, timestamp, runner, event_source)
+        VALUES (:name, :version, :timestamp, :runner, 'library run')"""
+    pg_client.execute_query(
+        query,
+        name=config.dataset.name,
+        version=config.dataset.version,
+        timestamp=config.execution_details.timestamp,
+        runner=config.execution_details.runner_string,
+    )
+
+
+def scrape_metadata(dataset: str) -> None:
+    """For a given recipes dataset, regenerates logs"""
+    logger.info(f"Re-scraping metadata for dataset {dataset}")
+    pg_client = postgres.PostgresClient(database=LOGGING_DB, schema=LOGGING_SCHEMA)
+
+    ## remove records in db for this recipe dataset
+    try:
+        pg_client.execute_query(
+            f"""
+            DELETE FROM {LOGGING_TABLE_NAME}
+            WHERE name = :dataset AND event_source='scrape';""",
+            dataset=dataset,
+        )
+    except Exception as e:
+        logger.warn(f"Could not delete from table: {e}")
+
+    ## scrape metadata, insert into table
+    versions = get_all_versions(dataset)
+    metadata = [
+        get_metadata(dataset, v).to_row()
+        for v in versions
+        #    if v not in ["testor", "2023_executive_points", "2017_adopted_points", "2017_adopted_polygons"]
+        #    and not (dataset == "doitt_buildingfootprints" and v == "20230414")
+        #    and not (dataset == "test_nypl_libraries")
```

Comment:
✂️?

Reply:
or trying to keep this around for later resolution of the datasets that didn't work?